graph/cycle_detection.cpp

```diff
@@ -0,0 +1,117 @@
+#include <vector>
+#include <algorithm>
+#include <iostream>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```